### PR TITLE
[MenuRoot] Remove wrong default value from docs

### PR DESCRIPTION
**Pull request by sai6855**
I was looking at source code of Menu and docs, [docs](https://deploy-preview-546--base-ui.netlify.app/base-ui/react-menu/components-api/#menu-root-props) says default value of `openOnHover` prop is `nested` but `nested` value is not defined in code and even if `openOnHover` prop is defined as `nested` following code (see line 27) would be illogical as `openOnHoverProp` will always be true. hence i removed default prop as `nested` from docs. 

https://github.com/mui/base-ui/blob/919d9ca513b3b2c52b175a33a858529408593a5f/packages/mui-base/src/Menu/Root/MenuRoot.tsx#L21-L30

### Before: 
image

### After
image

### Preview: 
https://deploy-preview-549--base-ui.netlify.app/base-ui/react-menu/components-api/#menu-root-props

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/mui-base/src/Menu/Root/MenuRoot.tsx [hunk 1]
```diff
@@ -139,8 +139,6 @@ namespace MenuRoot {
     delay?: number;
     /**
      * Whether the menu popup opens when the trigger is hovered after the provided `delay`.
-     *
-     * @default nested
```
Comment (michaldudak) [inline packages/mui-base/src/Menu/Root/MenuRoot.tsx:143]: I used to have such a prop in one of the previous implementation attempts and forgot to remove the `@default` annotation. We can add in plain text that by default `openOnHover` is set on nested menus.

Comment (sai6855) [reply packages/mui-base/src/Menu/Root/MenuRoot.tsx:143]: Updated https://deploy-preview-549--base-ui.netlify.app/base-ui/react-menu/components-api/#menu-root-props

File: packages/mui-base/src/Menu/Root/MenuRoot.tsx [hunk 2]
```diff
@@ -139,6 +139,7 @@ namespace MenuRoot {
     delay?: number;
     /**
      * Whether the menu popup opens when the trigger is hovered after the provided `delay`.
+     * By default, openOnHover is set to `true` for nested menus.
```
Comment (michaldudak) [inline packages/mui-base/src/Menu/Root/MenuRoot.tsx:142]:
```suggestion
     * By default, `openOnHover` is set to `true` for nested menus.
```

**Review by michaldudak — APPROVED** [review]

**Comment by michaldudak** [comment]
Thanks!

**Pull request merged**